### DISCUSSION TITLE
How to use transformImgUrl on img upload?

### DISCUSSION THREAD

**Discussion by CyberSekure** [imzbf/md-editor-v3#669] [Q&A] [answered]
Is there a way to use transformImgUrl when uploading an image via the MdEditor? I'm trying to store some temporary images locally as base64 before uploading to S3. Currently I'm getting the below, but want to display a shorter image name text.

![Screenshot 2024-09-12 at 17 23 28](https://github.com/user-attachments/assets/89f8f037-6fb4-4824-859c-a954e3b9e1c2)

Many thanks!

**Comment by imzbf** [comment, marked as answer]
If you need to display an image using base64, there seems to be no way to shorten it? 
When uploading an image, you can directly convert the image content in the `onUploadImg` event; there's no need to set `transformImgUrl`. This method is used when pasting content similar to `![](./img.png)`.

**Reply by CyberSekure** [reply]
Thank you, you prompted me to look into storing it in locally using URL.createObjectUrl. New to JS, vue and nuxt so thank you for your help
